plugin/src/main/java/org/opensearch/ml/action/agents/DeleteAgentTransportAction.java

```diff
@@ -30,41 +41,79 @@ public class DeleteAgentTransportAction extends HandledTransportAction<ActionReq
         ActionFilters actionFilters,
         Client client,
-        NamedXContentRegistry xContentRegistry
+        NamedXContentRegistry xContentRegistry,
+        ClusterService clusterService
     ) {
         super(MLAgentDeleteAction.NAME, transportService, actionFilters, MLAgentDeleteRequest::new);
         this.client = client;
         this.xContentRegistry = xContentRegistry;
+        this.clusterService = clusterService;
     }
 
     @Override
     protected void doExecute(Task task, ActionRequest request, ActionListener<DeleteResponse> actionListener) {
         MLAgentDeleteRequest mlAgentDeleteRequest = MLAgentDeleteRequest.fromActionRequest(request);
         String agentId = mlAgentDeleteRequest.getAgentId();
-        try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
-            ActionListener<DeleteResponse> wrappedListener = ActionListener.runBefore(actionListener, () -> context.restore());
-            DeleteRequest deleteRequest = new DeleteRequest(ML_AGENT_INDEX, agentId);
-            client.delete(deleteRequest, new ActionListener<DeleteResponse>() {
-                @Override
-                public void onResponse(DeleteResponse deleteResponse) {
-                    log.debug("Completed Delete Agent Request, agent id:{} deleted", agentId);
-                    wrappedListener.onResponse(deleteResponse);
-                }
+        GetRequest getRequest = new GetRequest(ML_AGENT_INDEX).id(agentId);
+        boolean isSuperAdmin = isSuperAdminUserWrapper(clusterService, client);
 
-                @Override
-                public void onFailure(Exception e) {
-                    log.error("Failed to delete ML Agent " + agentId, e);
-                    wrappedListener.onFailure(e);
+        try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+            client.get(getRequest, ActionListener.wrap(getResponse -> {
+                if (getResponse.isExists()) {
+                    try (XContentParser parser = createXContentParserFromRegistry(xContentRegistry, getResponse.getSourceAsBytesRef())) {
+                        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+                        MLAgent mlAgent = MLAgent.parse(parser);
+                        if (mlAgent.getIsHidden() && !isSuperAdmin) {
+                            actionListener
+                                .onFailure(
+                                    new OpenSearchStatusException(
+                                        "User doesn't have privilege to perform this operation on this hidden agent",
+                                        RestStatus.FORBIDDEN
+                                    )
+                                );
+                        } else {
+                            // If the agent is not hidden or if the user is a super admin, proceed with deletion
```

Comment:
looks like the agent is running without any access control. Anyone can delete an agent when he has the agent ID?

Reply:
Yeah, we don't have access control in agent yet.